### PR TITLE
Default AWS region to us-east-1 (#5211)

### LINKED ISSUE #3827: Cannot access Alibaba Cloud OSS via AmazonS3
**Describe the bug**

I use AmazonS3 to access OSS like this:
```
    AmazonS3Builder::new()
        .with_access_key_id(key_id)
        .with_secret_access_key(key_secret)
        .with_bucket_name(bucket)
        .with_url(endpoint)
        .build()
```
`endpoint` is `https://oss-cn-hangzhou.aliyuncs.com`,  this build failed with following error:

```
Failed to setup analytic engine: OpenObjectStore { source: Generic { store: "S3", source: UrlNotRecognised { url: "https://oss-cn-hangzhou.aliyuncs.com" } } }
```

This is a demo for using Python to access OSS via S3 client in https://www.alibabacloud.com/help/en/object-storage-service/latest/use-aws-s3-sdk-to-access-oss#section-jmf-a67-hat
```
import boto3
from botocore.config import Config
endpoint = 'https://oss-cn-hangzhou.aliyuncs.com'
access_key_id = 'OSS_AccessKeyId'
secret_access_key = 'OSS_AccessKeySecret'

s3 = boto3.client(
    's3',
    aws_access_key_id=access_key_id,
    aws_secret_access_key=secret_access_key,
    endpoint_url=endpoint,
    config=Config(s3={"addressing_style": "virtual", "signature_version": 's3v4'}))
```
I think there is something wrong in current S3 client implementation.

**To Reproduce**

See above
**Expected behavior**

**Additional context**

Build successfully.

### LINKED ISSUE #5211: [object-store] Make aws region optional.
We get this request upstream: https://github.com/pola-rs/polars/issues/13042

Where we cause an error because we try to infer the region. However it turns out any region would suffice because they have a private s3.

To be able to tailor to that use case I assume it would be better to make region optional. We can hack around this by just setting a region for the time being.

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Closes #5211
Closes #3827

# Rationale for this change
 

This has come up a number of times, and appears to cause friction for people using S3-compatible stores.

Most of such stores either support any region, or at the very least "us-east-1", and so we can default to `us-east-1`:

* https://github.com/minio/minio/discussions/15063
* https://developers.cloudflare.com/r2/api/s3/api/#bucket-region

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: object_store/src/aws/builder.rs [hunk 1]
```diff
@@ -559,19 +556,25 @@ impl AmazonS3Builder {
...
 
-    /// Set the AWS Access Key (required)
+    /// Set the AWS Access Key
     pub fn with_access_key_id(mut self, access_key_id: impl Into<String>) -> Self {
         self.access_key_id = Some(access_key_id.into());
         self
     }
 
-    /// Set the AWS Secret Access Key (required)
+    /// Set the AWS Secret Access Key
```
Comment (tustvold) [inline object_store/src/aws/builder.rs:565]: This is a drive-by fix as these are no longer required (we support other forms of credential)

File: object_store/src/aws/builder.rs [hunk 2]
```diff
@@ -559,19 +556,25 @@ impl AmazonS3Builder {
...
     pub fn with_secret_access_key(mut self, secret_access_key: impl Into<String>) -> Self {
         self.secret_access_key = Some(secret_access_key.into());
         self
     }
 
-    /// Set the region (e.g. `us-east-1`) (required)
+    /// Set the AWS Session Token to use for requests
+    pub fn with_token(mut self, token: impl Into<String>) -> Self {
```
Comment (tustvold) [inline object_store/src/aws/builder.rs:572]: This is moved up from lower down, to sit alongside the credentials it relates to. I also tweaked the doc to make it a bit clearer.

**Comment by tustvold** [comment]
FYI @ritchie46

**Comment by ritchie46** [comment]
Ah, nice one. Thanks for the heads up. Can remove my temporary workaround. ^^

**Review by alamb — APPROVED** [review]
LGTM -- thank you @tustvold

**Pull request merged**